scripts/install/opt/seagate/cortx/motr/libexec/motr_data_recovery.sh

```diff
@@ -444,11 +447,25 @@ replay_logs() {
             else
                 m0drlog "ERROR: Journal logs replay failed on local node"
                 (( exec_status|=1));
-            fi
+        fi
+
+        m0drlog "Get generation id of local node from segment 1"
+        LOCAL_SEG_GEN_ID="$($BECKTOOL -s $MD_DIR/m0d-$LOCAL_IOS_FID/db/o/100000000000000:2a -p)"
+        LOCAL_SEG_GEN_ID=$(echo $LOCAL_SEG_GEN_ID | cut -d "(" -f2 | cut -d ")" -f1)
+
+        if [[ $LOCAL_SEG_GEN_ID -eq 0 ]]; then
+            m0drlog "Get generation id of local node from segment 0"
+            LOCAL_SEG_GEN_ID="$($BECKTOOL -s $MD_DIR/m0d-$LOCAL_IOS_FID/db/o/100000000000000:29 -p)"
+            LOCAL_SEG_GEN_ID=$(echo $LOCAL_SEG_GEN_ID | cut -d "(" -f2 | cut -d ")" -f1)
+        fi
```

Comment:
"fi" this seem out of place, else is there below.

Reply:
Confusion seems to be due to github alignment.

Reply:
@madhavemuri: I'd suggest to add an example to https://jts.seagate.com/browse/EOS-11827.

Reply:
Looks like issue is with code itself.
@shashank-parulekar : Can you check the else path.

 456         if [[ $LOCAL_SEG_GEN_ID -eq 0 ]]; then
 457             m0drlog "Get generation id of local node from segment 0"
 458             LOCAL_SEG_GEN_ID="$($BECKTOOL -s $MD_DIR/m0d-$LOCAL_IOS_FID/db/o/100000000000000:29 -p)"
 459             LOCAL_SEG_GEN_ID=$(echo $LOCAL_SEG_GEN_ID | cut -d "(" -f2 | cut -d ")" -f1)
 460         fi
 461         echo "segment genid : ($LOCAL_SEG_GEN_ID)"  >  $MD_DIR/m0d-$LOCAL_IOS_FID/gen_id
 462         else
 463             m0drlog "ERROR: Mount failed! Can't replay journal logs on local node..."
 464             (( exec_status|=1));
 465         fi